### PR TITLE
add: `kiri`

### PR DESCRIPTION
**Pull request by MarcelRobitaille**
## Progress 

- [x] Edit packagelist
- [x] Add initial pacscript
- [ ] Contact devs
- [ ] Add maintainer to pacscript

### CODE REVIEW

File: packages/kiri-git/kiri-git.pacscript [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name=kiri-git
+repology=("project: kiri")
+homepage="https://github.com/leoheck/kiri"
+build_depends=("git" "python3-pip")
```
Comment (Henryws) [inline packages/kiri-git/kiri-git.pacscript:4]:
```suggestion
build_depends=("git" "python3-pip" "pkg-config")
```

**Review by Henryws — CHANGES_REQUESTED** [review]

File: packages/kiri-git/kiri-git.pacscript [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+depends=("git" "build-essential" "libgtk-3-dev" "pkg-config" "opam" "python-is-python3" "python3-pip" "kicad" "dos2unix" "coreutils" "zenity" "librsvg2-bin" "imagemagick" "xdotool" "rename")
```
[the remark this answers is not in the document]
Comment (MarcelRobitaille) [reply packages/kiri-git/kiri-git.pacscript:5]: The dependencies are taken directly from upstream.

- This software works with git, so it's a runtime dependency as well as a build dependency
- I agree that `build-essential` is odd. It's taken from upstream
- `pkg-config` is again taken from upstream. It's not required to build. I don't use it in the pacscript. I'm not sure if it's really needed upstream though.

[the remark this answers is not in the document]
Comment (Henryws) [reply packages/kiri-git/kiri-git.pacscript:5]: `git` is already installed with pacstall, same with `build-essential`, and my bad, `pkg-config` should stay in `build_depends`

[the remark this answers is not in the document]
Comment (MarcelRobitaille) [reply packages/kiri-git/kiri-git.pacscript:5]: @Henryws Maybe there was a miscommunication. My build script doesn't depend on `pkg-config`. It's listed as a dependency upstream.

File: packages/kiri-git/kiri-git.pacscript (hunk 1 above)
Comment (Henryws) [reply packages/kiri-git/kiri-git.pacscript]: This still needs to happen I believe.

File: packages/kiri-git/kiri-git.pacscript [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+arch=("any")
+provides=("kiri")
+conflicts=("kiri")
+pacdeps=("kicad-diff-git")
+url="https://github.com/leoheck/kiri.git"
+
+pkgver() {
+  git ls-remote "${url}" main | cut -f1 | cut -c1-8
+}
+version=$(pkgver)
```
Comment (Henryws) [inline packages/kiri-git/kiri-git.pacscript:18]:
```suggestion
version="$(pkgver)"
```

File: packages/kiri-git/kiri-git.pacscript [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+depends=("build-essential" "libgtk-3-dev" "pkg-config" "opam" "python-is-python3" "python3-pip" "kicad" "dos2unix" "coreutils" "zenity" "librsvg2-bin" "imagemagick" "xdotool" "rename")
+gives=("kiri")
+description="A tool for reviewing Schematics and Layouts of Git-versioned Kicad-projects, visually"
+maintainer="Marcel Robitaille <mail@marcelrobitaille.me>"
+arch=("any")
+provides=("kiri")
+conflicts=("kiri")
+pacdeps=("kicad-diff-git")
+url="https://github.com/leoheck/kiri.git"
+
```
Comment (Henryws) [inline packages/kiri-git/kiri-git.pacscript:14]:
```suggestion
```

File: packages/kiri-git/kiri-git.pacscript [hunk 5]
```diff
@@ -0,0 +1,36 @@
+name=kiri-git
+repology=("project: kiri")
+homepage="https://github.com/leoheck/kiri"
+build_depends=("python3-pip" "pkg-config")
+depends=("build-essential" "libgtk-3-dev" "pkg-config" "opam" "python-is-python3" "python3-pip" "kicad" "dos2unix" "coreutils" "zenity" "librsvg2-bin" "imagemagick" "xdotool" "rename")
+gives=("kiri")
+description="A tool for reviewing Schematics and Layouts of Git-versioned Kicad-projects, visually"
+maintainer="Marcel Robitaille <mail@marcelrobitaille.me>"
+arch=("any")
```
Comment (Henryws) [inline packages/kiri-git/kiri-git.pacscript:9]:
```suggestion
```

File: packages/kiri-git/kiri-git.pacscript [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  git submodule init
+  git -c protocol.file.allow=always submodule update
+}
+
+install() {
+  sudo mkdir -p "${pkgdir}/opt/kiri/"
+  sudo mkdir -p "${pkgdir}/usr/bin"
+  sudo cp -r ./* "${pkgdir}/opt/kiri"
+  for bin in "find_kicad5_sch_hier" "find_kicad6_sch_hier" "fix_svg_vias" "git-imgdiff" "kicad6_sheet_instances" "kicad_rev" "kicad_version" "kicad_version.py" "kicad_version_cliclick" "kicad_version_xdotool" "kiri" "kiri-file-picker" "kiri-server" "plot_kicad_sch" "plot_kicad_sch_macos" "plotgitsch_svg_tweaks"; do
+    sudo ln -s "../../opt/kiri/bin/${bin}" "${pkgdir}/usr/bin/${bin}"
```
Comment (Henryws) [inline packages/kiri-git/kiri-git.pacscript:34]: You can't use `ln` inside `install()`.[^1]

[^1]: https://github.com/pacstall/pacstall/wiki/Common-Pacscript-practices#handling-symlinks

**Review by Henryws — CHANGES_REQUESTED** [review]

**Comment by Elsie19** [comment]
Stale.

**Pull request closed without merging**